### PR TITLE
Add minimax m3vl

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?

### CODE REVIEW

File: src/transformers/models/doge/modeling_doge.py [hunk 1]
```diff
@@ -592,7 +592,6 @@ def forward(
             hidden_states = decoder_layer(
                 hidden_states,
                 attention_mask=causal_mask,
-                position_ids=position_ids,
```
Comment (vasqu) [inline src/transformers/models/doge/modeling_doge.py:595]: We still need to pass them here, top level needs it but below we can safely ignore

**Review by vasqu — APPROVED** [review]
Preapproval 🫡 nice work, we can refactor a bit afterwards but nothing major

Only revert the pos ids one since it could break some other models (e.g. doge)

File: src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py [hunk 2]
```diff
@@ -0,0 +1,1586 @@
...
+        pad = num_key_blocks * self.block_size - k_len
+
+        scores = torch.matmul(idx_q.float(), idx_k.float().transpose(-1, -2))
+        k_positions = torch.arange(k_len, device=idx_q.device)
+        token_future = k_positions[None, None, None, :] > position_ids[:, None, :, None]  # [B, 1, S_q, S_k]
+        scores = scores.masked_fill(token_future, float("-inf"))
+        if pad:
+            scores = F.pad(scores, (0, pad), value=float("-inf"))
+        scores = scores.view(batch, self.num_heads, q_len, num_key_blocks, self.block_size)
+        block_scores = scores.amax(dim=-1).amax(dim=1)  # -> [B, S_q, num_key_blocks]
```
Comment (nvpohanh) [inline src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: Why do we do `amax(dim=1)` here? does it mean that all GQA groups select the same blocks?

Comment (nvpohanh) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: This does not match SGLang's implementation: https://github.com/JustinTong0323/sglang/blob/334438dc6f230bf397797df56ac4ccbe531707a9/python/sglang/srt/layers/attention/minimax_sparse_ops/naive/flash_with_topk_idx.py#L42C54-L42C65

Comment (nvpohanh) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: @ArthurZucker could you check this? Thanks!

Comment (ArthurZucker) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: will have a look!

Comment (ArthurZucker) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: see #46719 this seem to have minimal effect!

Comment (ArthurZucker) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: patching today

Comment (nvpohanh) [reply src/transformers/models/minimax_m3_vl/modeling_minimax_m3_vl.py:580]: Thanks for fixing it!
